src/Sylius/Bundle/ApiBundle/StateProvider/RecursiveAdjustmentsStateProvider.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\StateProvider;
+
+use ApiPlatform\Metadata\Operation;
+use ApiPlatform\State\ProviderInterface;
+use Doctrine\Common\Collections\Collection;
+use Sylius\Component\Core\Model\AdjustmentInterface;
+use Sylius\Component\Order\Model\RecursiveAdjustmentsAwareInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+
+final readonly class RecursiveAdjustmentsStateProvider implements ProviderInterface
```

Comment:
We should be consistent with the suffix of state providers, because in one it is used `StateProvider` and in the other only `Provider`. I would be for change here and not using `State` keyword:
```suggestion
final readonly class RecursiveAdjustmentsProvider implements ProviderInterface
```